api/transport/handler.go

```diff
@@ -141,3 +163,19 @@ func DispatchOnewayHandler(
 
 	return h.HandleOneway(ctx, req)
 }
+
+// DispatchStreamHandler calls the stream handler, recovering from panics as
+// errors.
+func DispatchStreamHandler(
+	h StreamHandler,
+	stream ServerStream,
+) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			log.Printf("Stream handler panicked: %v\n%s", r, debug.Stack())
```

Comment:
We probably should thread the logger now that we have settled on injecting zap. I recognize this is copypasta from our existing panic handlers, and is probably fine regardless.

Reply:
going to leave this for now, these Dispatch functions are more recommendations for the transports than requirements, we can easily switch them out with better APIs if we want (maybe more of a 'pkg' function)